shared/corefxlab/System.IO.Pipelines/ReadableBufferReader.cs

```diff
@@ -43,11 +43,6 @@ public ReadCursor Cursor
             {
                 var part = _enumerator.Current;
 
-                if (_end)
-                {
-                    return new ReadCursor(part.Segment, part.Start + _currentSpan.Length);
```

Comment:
This is really strange, now when I look at it.

Reply:
The assumption was that end = end of the current segment. Turns out it can also mean beginning of an empty segment.